ls1intum/Artemis
# Feature/shared/generic button component

### Checklist

### Motivation and Context



Implement a generic button component: https://github.com/ls1intum/Artemis/issues/817

### Description


Simple button component that has a loading & disabled state.
Supports both use with text and/or icon.

I replaced the existing buttons in #/exercise/5/participation for programming-exercises at the top right.

The SharedComponentModule is not an import/export of the SharedModule as this could create cyclic dependencies (the generic components need exports of the SharedModule). Therefore it must be imported by modules that want to use it.

### Steps for Testing


1. Log in to Artemis
2. Go to #/exercise/`exerciseId`/participation for a programming exercise
3. Check buttons at the top right and make sure that they still work (e.g. click)

### Screenshots



**From the participations list:**

![image](https://user-images.githubusercontent.com/28230611/64712536-aee92a80-d4bb-11e9-93ff-a048e5caf147.png)

**Example without icon:**

![image](https://user-images.githubusercontent.com/28230611/64718696-080a8b80-d4c7-11e9-8597-94143d481687.png)

**Example without title:**

![image](https://user-images.githubusercontent.com/28230611/64718759-22dd0000-d4c7-11e9-825c-fb429c6c5b95.png)
